pkg/daemon/daemon.go

```diff
@@ -1648,79 +1648,79 @@ func PersistNetworkInterfaces(osRoot string) error {
 		nmstateBinary = filepath.Join(originalContainerBin, "nmstatectl")
 	}
 
-	cmd := exec.Command(nmstateBinary, "persist-nic-names", "--root", osRoot)
+	// For the moment, we only look at RHEL-like systems...this logic isn't
+	// yet aiming to try to handle Fedora-level updates.  For that, most
+	// likely this NIC pinning should actually be driven automatically by
+	// host updates.  If you change this, you'll need to change the conditions
+	// below too.
+	persisting := hostos.IsEL8()
+	cleanup := hostos.IsEL9()
+	if !(persisting || cleanup) {
+		return nil
+	}
 
-	if hostos.IsEL8() {
-		logSystem("Persisting NIC names for RHEL8 host system")
+	tmpKargs, err := os.CreateTemp("", "nmstate-kargs")
+	if err != nil {
+		return err
+	}
+	defer os.Remove(tmpKargs.Name())
 
-		// Workaround for https://issues.redhat.com/browse/OCPBUGS-16035 .
-		// Create /etc/systemd/network directory if it doesn't exist.
-		dirPath := filepath.Join(osRoot, systemdNetworkDir)
-		if err := os.MkdirAll(dirPath, defaultDirectoryPermissions); err != nil {
-			return fmt.Errorf("failed to create directory %s: %w", dirPath, err)
-		}
+	cmd := exec.Command(nmstateBinary, "persist-nic-names", "--root", osRoot, "--kargs-out", tmpKargs.Name())
 
-		// nmstate always logs to stderr, so we need to capture/forward that too
-		cmd.Stdout = os.Stdout
-		cmd.Stderr = os.Stderr
-		klog.Infof("Running: %s", strings.Join(cmd.Args, " "))
-		if err := cmd.Run(); err != nil {
-			return fmt.Errorf("failed to run nmstatectl: %w", err)
-		}
-	} else if hostos.IsEL9() {
-		ifnames, err := getIfnamesFromLinkFiles(osRoot)
-		if err != nil {
-			klog.Warningf("Failed to persist NIC names: %v", err)
-			return nil
-		}
-
-		ifnamesKeys := []string{}
-		for ifname := range ifnames {
-			ifnamesKeys = append(ifnamesKeys, ifname)
-		}
-		if len(ifnamesKeys) > 0 {
-			logSystem("Have persisted ifnames for %s", strings.Join(ifnamesKeys, ", "))
-		}
+	if persisting {
+		klog.Info("Persisting NIC names for RHEL8 host system")
+	} else if cleanup {
+		cmd.Args = append(cmd.Args, "--cleanup")
+	} else {
+		// Should be unreachable
+		panic("Unexpected host OS")
```

Comment:
I understand that we won't reach hear most likely but still maybe we should error out instead of panic.

Reply:
Changed to error like:

```
return fmt.Errorf("Unexpected host OS %s", hostos.ToPrometheusLabel())
```